rudder-core/src/main/scala/com/normation/rudder/domain/reports/bean/ExecutionBatch.scala

```diff
@@ -489,36 +485,57 @@ case class ConfigurationExecutionBatch(
-       val componentvalues = directive.flatMap{ nodestatus =>
-         val components = nodestatus.components.filter(_.component==id)
-         getComponentValuesRuleStatus(directiveid,id,component.groupedComponentValues,components) ++
+
+    // Make a map of all component, currently we cannot use class ReportComponent correctly since it does not bbehave correctly
+    // We want to have a map that give for a Component, which unexpanded value is set, and for which unexpanded value, which values are expected
+    val componentMap = components.
+      // Group by component
+      groupBy(_.componentName).
+        // transform them in couple (value,unexpandedValue)
+        mapValues(_.flatMap(_.groupedComponentValues).
+            // Group values with the same unexpandedValue together
+            groupBy(_._2).
+              // Make a set of expected value so we dont have them twice
+              mapValues(_.map(_._1).toSet))
+
+     (componentMap.map{ case (id,groupedComponentValues) =>
+       val componentvalues = directive.flatMap{ directiveReports =>
+         val components = directiveReports.components.filter(_.component==id)
+         getComponentValuesRuleStatus(directiveid,id,groupedComponentValues,components) ++
          getUnexpectedComponentValuesRuleStatus(directiveid,id,components.flatMap(_.unexpectedCptValues))
        }
        ComponentRuleStatusReport(directiveid,id,componentvalues)
-     }
+     }).toSeq
  }
   /**
    * Get the status of expected values of the component passed as a parameter
    * Parameters:
    * directiveId : Values we are looking for are contained in that directive
    * component   : Values we are looking for are contained in that component
-   * values      : Expected values format
+   * values      : Expected values format, a map of expected unexpanded value => real value
    * components  : Latest components report
    */
- def getComponentValuesRuleStatus(directiveid:DirectiveId, component:String, values:Seq[(String, Option[String])], components:Seq[ComponentStatusReport]) : Seq[ComponentValueRuleStatusReport]={
-     values.map{
-       case (value, unexpanded) =>
-         val componentValues = components.flatMap(_.componentValues.filter(_.componentValue==value))
-         val nodes = componentValues.map(value => NodeReport(value.nodeId,value.reportType,value.message))
+ def getComponentValuesRuleStatus(directiveid:DirectiveId, component:String, values:Map[Option[String], Set[String]], components:Seq[ComponentStatusReport]) : Set[ComponentValueRuleStatusReport]={
+
+   (values.flatMap{
+     case (unexpanded,values) =>
+       // Get all reports that match a value from our unexpandedValue, and group them by value
+       val componentValues = components.flatMap(_.componentValues.filter(v => values.contains(v.componentValue))).groupBy(_.componentValue)
+       for {
+         // for each value, process a reporting date
+         (value,values) <- componentValues
```

Comment:
what about those that don't match ?
plus, you are using twice values in 3 lines, so when rereading, it's hard to know which one we use